### DISCUSSION TITLE
Is it possible to use LazyVim utils in my own config?

### DISCUSSION THREAD

**Discussion by dotfrag** [LazyVim/LazyVim#3257] [Q&A] [answered]
Can I somehow use LazyVim utils in my own config without having to use submodules and clone the whole thing? Maybe some fancy lazy import?

I've been using LazyVim for so long that it feels lacking to use my personal nvim config without a significant portion of LazyVim utils and I'd rather not shamefully copy-paste and manually update the files in my repo.

**Comment by folke** [comment, marked as answer]
You can just install the Lazyvim plugin without adding the import.
Then you can use all the utisl

**Comment by dotfrag** [comment]
That worked, I was having some bad required statements..

Is there an easy way of making `require('lazyvim.utils')` global like in LazyVim? I tried adding it before lazy setup, but of course it isn't loaded yet. If I add it after, then the plugins that need it are loaded earlier. I tried adding it in the `config` key of LazyVim plugin but still doesn't work.

**Comment by dotfrag** [comment]
I think the issue wasn't with my require but with my setup not actually loading LazyVim, and then LazyVim fails to resolve the global:

```text
Failed to load `plugins.lualine`

...cal/share/nvim/lazy/LazyVim/lua/lazyvim/util/lualine.lua:135: attempt to index global 'LazyVim' (a nil value)

# stacktrace:
  - /LazyVim/lua/lazyvim/util/lualine.lua:135 _in_ **root_dir**
  - ~/.config/nvim/lua/plugins/lualine.lua:27
  - ~/.config/nvim/init.lua:38
 ```

**Reply by dpetka2001** [reply]
That's because `LazyVim` is used as global variable for `require("lazyvim.util")`. Since you're not loading the plugin you'll have to replace all occurences of `LazyVim` with `require("lazyvim.util")` in your config.

**Reply by dotfrag** [reply]
I'm already trying that, but I think the problem is that the thing I'm requiring, i.e. `require('lazyvim.util').lualine.pretty_path()` cannot resolve LazyVim itself, e.g. https://github.com/LazyVim/LazyVim/blob/8e67bb124009e0dbf29f8e78d701d07756670278/lua/lazyvim/util/lualine.lua#L93

EDIT: The only way for the above to work, is if I add `_G.LazyVim = require('lazyvim.util')` to the top of the plugin file (i.e. `lua/plugins/lualine.lua`), but that means I have to add it everywhere it's needed, hence why I'm asking if it's possible to add it as a true global.

**Reply by dpetka2001** [reply]
Have you tried adding that at the top of your root `init.lua`?

**Reply by dotfrag** [reply]
Yes, if I add it to the very top, it fails to find the module, probably because lazy has not been setup yet. And I cannot add it after lazy setup either, because the plugins won't know about it.

**Reply by dotfrag** [reply]
I got it working by manually adding the path before lazy setup:

```lua

local lazyvimpath = vim.fn.stdpath("data") .. "/lazy/LazyVim"
vim.opt.rtp:prepend(lazyvimpath)
_G.LazyVim = require("lazyvim.util")
```

**Reply by dpetka2001** [reply]
Nice!! Thanks for sharing this :smile:
